associacao-brasileira-de-normas-tecnicas-ufrgs-initials.csl

```diff
@@ -245,105 +348,134 @@
+        </else-if>
+        <else-if type="article-journal paper-conference" match="any">
+          <choose>
+            <if variable="status" match="any">
+              <text variable="status" font-style="italic"/>
+            </if>
+            <else>
+              <text value="s. d." font-style="italic" prefix="[" suffix="]"/>
+            </else>
+          </choose>
+        </else-if>
+        <else>
+          <text value="s. d." font-style="italic" prefix="[" suffix="]"/>
+        </else>
+      </choose>
+    </group>
+  </macro>
+  <macro name="issued-year">
+    <!-- Macro responsável por mostrar apenas o ano -->
     <choose>
-      <if variable="issued" match="any">
+      <if variable="event-date" match="any">
+        <date date-parts="year" form="numeric" variable="event-date" prefix=" "/>
+      </if>
+      <else-if is-uncertain-date="issued">
+        <date variable="issued">
+          <date-part name="year" prefix="[" suffix="]"/>
+        </date>
+      </else-if>
+      <else-if variable="issued" match="any">
+        <date date-parts="year" form="numeric" variable="issued" prefix=" "/>
+      </else-if>
+      <else-if type="article-journal">
         <choose>
-          <if type="patent song post-weblog article-newspaper personal_communication" match="any">
-            <date variable="issued" prefix=" " suffix=".">
-              <date-part name="day" suffix=" "/>
-              <date-part name="month" form="short" suffix=" "/>
-              <date-part name="year"/>
-            </date>
+          <if variable="status" match="any">
+            <text variable="status" font-style="italic"/>
           </if>
           <else>
-            <date variable="issued" suffix=".">
-              <date-part name="year" prefix=" "/>
-            </date>
+            <text value="s. d." font-style="italic" prefix="[" suffix="]"/>
           </else>
         </choose>
-      </if>
-      <else>
-        <text value=" [s. d.]" font-style="italic"/>
-      </else>
-    </choose>
-  </macro>
-  <macro name="issued-year">
-    <choose>
-      <if variable="issued" match="any">
-        <date variable="issued">
-          <date-part name="year"/>
-        </date>
-      </if>
-      <else-if type="paper-conference" match="all">
-        <date date-parts="year" form="numeric" variable="issued">
-          <date-part name="year"/>
-        </date>
       </else-if>
       <else>
-        <text value=" [s. d.]" font-style="italic"/>
+        <text value="s. d." font-style="italic" prefix="[" suffix="]"/>
       </else>
     </choose>
   </macro>
   <macro name="issued-legislation">
+    <!-- Macro responsável por mostrar a data para documentos legislativos -->
     <date variable="issued">
       <date-part name="day" suffix=" "/>
       <date-part name="month" form="short" suffix=". " text-case="lowercase"/>
-      <date-part name="year"/>
+      <date-part name="year" prefix=" "/>
```

Comment:
the space here isn't necessary (since month has a space in the suffix). Same for line 474 below

Reply:
Even when there is no month?

Reply:
I didn't look through the whole code in great detail, but if there's no space before the year without a month, there would also not be a space before the full date and you should add a prefix with a space to the whole date (i.e. in line 463 instead of here)

Reply:
Thanks!